sympy/core/mul.py

```diff
@@ -1264,26 +1266,38 @@ def _eval_is_odd(self):
         is_integer = self.is_integer
 
         if is_integer:
-            r, acc = True, 1
+            #The entire expression is an integer, but it
+            #can be decomposed into fractions (i.e.
+            #numerator and denominator.
+            #The counters keep track of multiples of 2 in
+            #numerator and denominator to determine oddness.
+            count_denom = 0
+            count_num = 0
             for t in self.args:
-                if not t.is_integer:
-                    return None
-                elif t.is_even:
-                    r = False
-                elif t.is_integer:
-                    if r is False:
-                        pass
-                    elif acc != 1 and (acc + t).is_odd:
-                        r = False
-                    elif t.is_odd is None:
-                        r = None
-                acc = t
-            return r
+                if t == S.Half:
+                    count_denom += 1
+                elif isinstance(t,Symbol):
+                    if t.is_integer:
+                        if t.is_even:
+                            count_num += 1
+                        elif t.is_odd is None or t.is_even is None:
+                            return None
+                                               
+                elif isinstance(t,int):
```

Comment:
Can you clarify what you mean by this?  My reasoning behind this piece is that if t isn't a fraction (i.e. is_integer) then that means it's part of the numerator.  If it's even (has, at the minimum, a factor of 2 in it), then we know there is a 2 in the numerator, so the count_num +=1.  For the next elif, if any part of the expression is unknown to be odd or even, then we can't evaluate it, so None.

For the bottom elif isinstance, is int not a SymPy type?  Should it be Integer instead?
Hope I explained my reasoning a bit!  Let me know if I should clarify more or any other suggestions.